examples/azure/client_factory.md

```diff
@@ -0,0 +1,133 @@
+import (
+    "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+)
+
+func SomeMethod() {
+    subscriptionID := "your subscription ID"
+    baseURI := "management.azure.com"
+
+    // Create a VM client and return
+	vmClient := compute.NewVirtualMachinesClientWithBaseURI(baseURI, subscriptionID)
+
+    // Use client / etc
+}
+```
+
+Now we have code that can be used with any cloud environment just by changing the base URI we pass to the clients (Public is shown in above example). 
+
+## Environment URI Lookups
+
+Rather than hardcode these URI's, they should be looked up from an authoritiative source. Fortunately, the Autorest-GO library (used by the Go SDK) provides such functionality. The `client_factory` module makes use of the Autorest `EnvironmentFromName(envName string)` function to return the appropriate structure. 
+
+This Environment structure has the following fields configured for each cloud environment:
+
+```golang
+ManagementPortalURL          
+PublishSettingsURL           
+ServiceManagementEndpoint    
+ResourceManagerEndpoint      
+ActiveDirectoryEndpoint      
+GalleryEndpoint              
+KeyVaultEndpoint             
+GraphEndpoint                
+ServiceBusEndpoint           
+BatchManagementEndpoint      
+StorageEndpointSuffix        
+SQLDatabaseDNSSuffix         
+TrafficManagerDNSSuffix      
+KeyVaultDNSSuffix            
+ServiceBusEndpointSuffix     
+ServiceManagementVMDNSSuffix 
+ResourceManagerVMDNSSuffix   
+ContainerRegistryDNSSuffix   
+CosmosDBDNSSuffix            
+TokenAudience                
+APIManagementHostNameSuffix  
+SynapseEndpointSuffix        
+ResourceIdentifiers          
+```
+
+Using these URI's, `client_factory` is able to instantiate and return properly configured SDK clients without module developers having to duplicate this code.
+
+## Configuration and Defaults
+
+To configure different cloud environments, we will use the same `AZURE_ENVIRONMENT` environment variable that the Go SDK uses. This can be set to one of the following values:
+
+|Value                      |Cloud Environment  |
+|---------------------------|-------------------|
+|"AzureChinaCloud"          |ChinaCloud         |
+|"AzureGermanCloud"         |GermanCloud        |
+|"AzurePublicCloud"         |PublicCloud        |
+|"AzureUSGovernmentCloud"   |USGovernmentCloud  |
+|"AzureStackCloud"          |Azure stack        |
+
+Note that when using the "AzureStackCloud" setting, you must also set the `AZURE_ENVIRONMENT_FILEPATH` variable to point to a JSON file containing your Azure Stack URI deatils.
+
+>NOTE: The default behavior of the `client_factory` is to use the AzurePublicCloud environment. This requires no work from the developer to configure, and ensures consistent behavior with the current SDK code.       
+
+## Usage Patterns
+
+Modules authors will interact with the `client_factory` through the `NewClientFactory` method on the `azure` package as shown below:
+
+```golang
```

Comment:
One advantage of moving these docs into the Terratest docs site is that you can use `include` to pull in code snippets from real, tested code, rather than copy/pasting that code directly into Markdown. See the `{% include ... %}` usage here, for example: https://github.com/gruntwork-io/terratest/blob/master/docs/_docs/01_getting-started/quick-start.md#example-1-terraform-hello-world. That `include` gets rendered as follows: https://terratest.gruntwork.io/docs/getting-started/quick-start/#example-1-terraform-hello-world

Reply:
Will revisit on outcome of moving this into Jekyll site (above).